h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -128,6 +128,23 @@ protected ModelBuilder(P parms, boolean startup_once, String externalSchemaDirec
   /** gbm -> "hex.schemas." ; custAlgo -> "org.myOrg.schemas." */
   public static String schemaDirectory(String urlName) { return SCHEMAS[ArrayUtils.find(ALGOBASES,urlName)]; }
 
+  /**
+   * Helper method that enables preprocessing with TE model. It's user's responsibility to make sure TE is trained in an appropriate way to avoid data leakage.
```

Comment:
It's definitely true right now it's up to the user to provide a correct TE model, but we can do a lot of validations - we might reject a target encoder provided by a user if there are obvious mismatches, e.g. target variable type.

Such validations can definitely be done.